packages/webview_flutter/webview_flutter_platform_interface/lib/v4/src/types/navigation_callback_creation_params.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+/// Object specifying creation parameters for creating a [NavigationCallbackDelegate].
+///
+/// Platform specific implementations can add additional fields by extending
+/// this class.
+///
+/// {@tool sample}
+/// This example demonstrates how to extend the [NavigationCallbackCreationParams] to
+/// provide additional platform specific parameters.
+///
+/// When extending [NavigationCallbackCreationParams] additional
+/// parameters should always accept `null` or have a default value to prevent
+/// breaking changes.
+///
+/// ```dart
+/// class AndroidNavigationCallbackCreationParams extends NavigationCallbackCreationParams {
+///   AndroidNavigationCallbackCreationParams._(
+///     // This parameter prevents breaking changes later.
+///     // ignore: avoid_unused_constructor_parameters
+///     NavigationCallbackCreationParams params, {
+///     this.filter,
+///   }) : super();
+///
+///   factory AndroidNavigationCallbackCreationParams.fromNavigationCallbackCreationParams(
+///       NavigationCallbackCreationParams params, {
+///       String? filter,
+///   }) {
+///     return AndroidNavigationCallbackCreationParams._(params, filter: filter);
+///   }
+///
+///   final String? filter;
+/// }
+/// ```
+/// {@end-tool}
+@immutable
+class NavigationCallbackCreationParams {
```

Comment:
`NavigationCallbackHandlerCreationParams`

Reply:
Renamed to `PlatformNavigationCallbackDelegateCreationParams` to match the `PlatformNavigationCallbackDelegate`.